### PR TITLE
Adjust crash type ignore and threshold

### PR DESCRIPTION
**Pull request by Navidem**

### CODE REVIEW

File: src/appengine/handlers/cron/grouper.py [hunk 1]
```diff
@@ -171,6 +174,7 @@ def _group_testcases_based_on_variants(testcase_map):
 
     # Determine anomalous candidates.
     threshold = VARIANT_THRESHOLD_PERCENTAGE * project_num_testcases[project]
+    threshold = min(threshold, VARINAT_MAX_THRESHOLD)
```
Comment (oliverchang) [inline src/appengine/handlers/cron/grouper.py:177]: Can we just simplify the below and do:

```
# Clamp threshold. 
threshold = min(threshold, VARINAT_MAX_THRESHOLD)
threshold = max(VARIANT_MIN_THRESHOLD, threshold)
```

Then we can delete that if statement on line 179.

File: src/appengine/handlers/cron/grouper.py [hunk 2]
```diff
@@ -32,9 +32,12 @@
...
 GROUP_MAX_TESTCASE_LIMIT = 25
 
-VARIANT_CRASHES_IGNORE = ['Out-of-memory', 'Timeout']
+VARIANT_CRASHES_IGNORE = [
+    'Out-of-memory', 'Timeout', 'Missing-library', 'Data race'
+]
 VARIANT_THRESHOLD_PERCENTAGE = 0.2
-VARIANT_MIN_THRESHOLD = 10
+VARIANT_MIN_THRESHOLD = 5
+VARINAT_MAX_THRESHOLD = 10
```
Comment (oliverchang) [inline src/appengine/handlers/cron/grouper.py:40]: nit: fix spelling of variant

**Review by oliverchang — APPROVED** [review]
i fixed my comments myself.

File: src/appengine/handlers/cron/grouper.py [hunk 3]
```diff
@@ -32,9 +33,12 @@
 
 GROUP_MAX_TESTCASE_LIMIT = 25
 
-VARIANT_CRASHES_IGNORE = ['Out-of-memory', 'Timeout']
+VARIANT_CRASHES_IGNORE = re.compile(
+    r'^Out-of-memory|^Timeout|^Missing-library|^Data race')
```
Comment (oliverchang) [inline src/appengine/handlers/cron/grouper.py:37]: nit: do

```
r'^(Out-of-memory|...|...)'
```

to avoid many repetitions of `^`.

Comment (Navidem) [reply src/appengine/handlers/cron/grouper.py:37]: Oh! nice!

**Review by oliverchang — APPROVED** [review]

**Pull request merged**